test/giffy_dialog_test.dart

```diff
@@ -1,39 +1,136 @@
-  testWidgets('Check button smoke test', (WidgetTester tester) async {
-    provideMockedNetworkImages(() async {
-      // Build our app and trigger a frame.
-      await tester.pumpWidget(new MyApp());
+  group('Dialog Assertion Test', () {
+    testWidgets(
+        'AssetGiffyDialog throws if initialized with null image and title',
+        (WidgetTester tester) async {
+      try {
+        await tester.pumpWidget(AssetGiffyDialog(
+          onOkButtonPressed: () {},
+          image: null,
+          title: null,
+        ));
+      } catch (error) {
+        expect(error, isAssertionError);
+      }
+    });
+
+    testWidgets(
+        'FlareGiffyDialog throws if initialized with null flarePath, flareAnimation and title',
+        (WidgetTester tester) async {
+      try {
+        await tester.pumpWidget(FlareGiffyDialog(
+          onOkButtonPressed: () {},
+          flarePath: null,
+          title: null,
+          flareAnimation: null,
+        ));
+      } catch (error) {
+        expect(error, isAssertionError);
+      }
+    });
+
+    testWidgets(
+        'NetworkGiffyDialog throws if initialized with null image and title',
+        (WidgetTester tester) async {
+      try {
+        await tester.pumpWidget(NetworkGiffyDialog(
+          onOkButtonPressed: () {},
+          image: null,
+          title: null,
+        ));
+      } catch (error) {
+        expect(error, isAssertionError);
+      }
+    });
+  });
+
+  group('Dialog Smoke and Orientation Test', () {
+    testWidgets('Check Portrait Dialog test', (WidgetTester tester) async {
+      provideMockedNetworkImages(() async {
+        final TestWidgetsFlutterBinding binding =
+            TestWidgetsFlutterBinding.ensureInitialized();
+        await binding.setSurfaceSize(Size(PORTRAIT_WIDTH, PORTRAIT_HEIGHT));
+
+        // Build our app and trigger a frame.
+        await tester.pumpWidget(new MyApp());
+
+        expect(find.byKey(keys[0]), findsOneWidget);
+
+        await tester.tap(find.byKey(keys[0]));
+        await tester.pump();
+
+        expect(find.byKey(keys[1]), findsOneWidget);
+
+        await tester.tap(find.text('Cancel'));
+        await tester.pump();
+
+        expect(find.byKey(keys[2]), findsOneWidget);
+
+        await tester.tap(find.byKey(keys[2]));
+        await tester.pump();
+
+        expect(find.byKey(keys[3]), findsOneWidget);
+
+        await tester.tap(find.text('Cancel'));
+        await tester.pump();
+
+        expect(find.byKey(keys[4]), findsOneWidget);
+
+        await tester.tap(find.byKey(keys[4]));
+        await tester.pump();
+
+        expect(find.byKey(keys[5]), findsOneWidget);
+      });
+    });
+
+    testWidgets('Check Landscape Dialog test', (WidgetTester tester) async {
+      provideMockedNetworkImages(() async {
+        final TestWidgetsFlutterBinding binding =
+            TestWidgetsFlutterBinding.ensureInitialized();
+        await binding.setSurfaceSize(Size(LANDSCAPE_WIDTH, LANDSCAPE_HEIGHT));
```

Comment:
```suggestion
        binding.window.physicalSizeTestValue = Size(PORTRAIT_WIDTH, PORTRAIT_HEIGHT);
```
MediaQuery use `window.physicalSize` to obtain the device size

Reply:
I think that is the reason which cause you test could not covering the `Orientation.portrait` case.

Reply:
@boyan01 It didn't work, the tests are failing, can you please have a look ?

`Expected: exactly one matching node in the widget tree
  Actual: ?:<zero widgets with key [<'FlareDialog'>] (ignoring offstage widgets)>
   Which: means none were found but one was expected
`

Reply:
increase the screen size can fix it
```dart
const double PORTRAIT_WIDTH = 1800.0;
const double PORTRAIT_HEIGHT = 2400.0;
const double LANDSCAPE_WIDTH = 2400.0;
const double LANDSCAPE_HEIGHT = 1800.0;
```

Reply:
@boyan01 The test got passed, but it still didn't cover the `Orientation.portrait` case.

Reply:
> MediaQuery use `window.physicalSize` to obtain the device size

Do we really need to change this size parameter here? As flutter runs the test in landscape mode by default.

Reply:
> > MediaQuery use `window.physicalSize` to obtain the device size
> 
> Do we really need to change this size parameter here? As flutter runs the test in landscape mode by default.

sorry, my bad.  I just paste it into wrong line, it maybe only need change the size when run `Check Portrait Dialog test`

Reply:
Hey, it got fixed. Thanks again for your help.

[link](https://codecov.io/gh/xsahil03x/giffy_dialog/branch/increase_codecov)